python_modules/dagster-graphql/dagster_graphql_tests/graphql/test_assets.py

```diff
@@ -260,3 +291,58 @@ def test_asset_asof_timestamp(self, graphql_context):
         materializations = result.data["assetOrError"]["assetMaterializations"]
         assert len(materializations) == 1
         assert first_timestamp == int(materializations[0]["materializationEvent"]["timestamp"])
+
+    def test_asset_in_progress_foo(self, graphql_context):
```

Comment:
This test case takes a very long time to run (>5 seconds per execution context, because the database needs to be started). Also requires persistent Dagster instance because it creates one thread to run a hanging asset, and one thread to execute graphql query and kill the run. Is there any way to do this more gracefully?

Reply:
Thanks, going to change test cases based on `test_run_cancellation.py`